### DISCUSSION TITLE
Is it possible to use code generation to only generate a case class from a defined Table?

### DISCUSSION THREAD

**Discussion by Daniel-Khodabakhsh** [slick/slick#2299] [Questions] [answered]
Is it possible to use the `slick-codegen` library to generate only the table's case classes, given some table definitions?
I tried looking around for examples but didn't see any. It looks like it might be possible with code generation customisation but just wanted to ask before I dove deeper.

**Comment by nafg** [comment, marked as answer]
It might be easier to achieve with https://github.com/nafg/slick-additions/tree/master/slick-additions-codegen/src/main/scala/slick/additions/codegen

It's an alternative code generator, based on scalameta instead of strings, and designed to be flexible.

**Reply by nafg** [reply]
I should mention that I use it that way. I have a project where I generate the table definitions into a backend-only module and the case classes into a shared module.

**Reply by nafg** [reply]
Actually an example of doing that is at the end of the github readme: https://github.com/nafg/slick-additions#slick-additions-codegen
